### ISSUE TITLE
facet background load not working

### ISSUE THREAD

**Issue by williamhoos** [clinwiki-org/clinwiki#110]
if facets are more than 10, there is some background load but to trigger, it requires opening the facet dropdown, then clicking on a study and returning to the search at which time the longer list which scrolls in the dropdown loads. Doesn't ever load if staying on the search page. Sometimes hangs with the animated loading dots after returning from a study.

**Comment by williamhoos** [comment]
seems a little inconsistent. See video https://screencast.com/t/cYHbB6vv9j9

Most facets load but state and city seem to hang and not show past first list. If I change page (back or to a study) and then return, it does load. Almost seems like once it has loaded the full list once then it stays loaded and updating as searches change. 

Could it be something that if a list hasn't finished loading in some time then it stops trying to load the list until a page change?

Sample search: https://clinwiki-prod.herokuapp.com/search/ROS1ders

**Comment by relwell** [comment]
It looks like what's happening is that the pre-loaded buckets never get replaced or appended to when you can add more data to the resultset. Probably a frontend issue. Looking into it now.

**Comment by relwell** [comment]
I've pushed a change that seems to fix the issue. There are a few things at play here:

1) The approach we're taking to updating components in React is starting to get out of date. Instead of driving requests through actions, we should be using Sagas for handling modification to the Redux store. This helps make sure that state changes get cascaded down to child components correctly. Making that change is a pretty significant refactor. Maybe we want to do around the same time as adding end to end testing, since that usually needs refactoring anyway. For now, I've forced the component to update at the end of the action. It's not perfect, but it seems to be working.

2) A lot of our gems were out of date on the backend, and upgrading them ended up being pretty non-trivial, but required as part of the investigation. It's worth noting that for now we want to pin to Searchkick 2.3.0, even though there's a newer version, because there are breaking changes that affect how our application presently works.
